### DISCUSSION TITLE
improvement(blob/service): revisit blob.GetAll

### DISCUSSION THREAD

**Discussion by vgonkivs** [celestiaorg/celestia-node#3245] [Ideas]
During the initial implementation of the blob service, we allow users to request blobs under multiple namespaces at the same time. It works well in successful cases. However, we haven't considered how users should handle both positive and negative cases at the same time(when the result contains found blobs and errors). Also, the returned error can contain errors from multiple requests which makes handling them more tricky. I propose to limit the request to one namespace, so the user will have a clear response - success or fail.

**Comment by renaynay** [comment]
I am for a cleaner solution which is a `GetAll` for 1 namespace only. I believe during the initial implementation of `BlobModule`, I also brought up this point, but I am in favour of simplicity (1 request for 1 namespace)

**Comment by Wondertan** [comment]
Usually, I prefer APIs that strive to cover more use cases yet balance that with simplicity. The original idea here was to enable a simple way for rollups to get data from multiple namespaces, like in case header and block data are separated, and it would be great to get some feedback from rollups on whether this is still needed. 

We also have now da interface integrated that aims for better devex while blobmodule could be optimized for our interests/needs. And I think this should be informed by the lower level protocol. For example, the blobulus workstream allows us to not pass namespaces for blob by commitment retrieval. Also, the interfaces should allow to squeeze out max performance out of the lower level protocol. Considering all that, I would return to this discussion when we have Blobulus workstream implemented.
